addons/mail/static/src/components/message_list/message_list.js

```diff
@@ -523,6 +546,20 @@ export class MessageList extends Component {
      * @param {ScrollEvent} ev
      */
     onScroll(ev) {
+        if (this.threadView) {
+            const isAtBottom = (
+                this._getScrollableElement().scrollTop
+                    >=
+                    this._getScrollableElement().scrollHeight
+                    -
+                    this._getScrollableElement().clientHeight
+                    -
+                    30
+            );
```

Comment:
I think this should be single line even if it's long, or maybe with 2 or 3 lines max, but this takes too much space and doesn't help readability, quite the contrary.

Reply:
Also `hasAutoScrollOnMessageReceived` should use this new variable, it does exactly the same computation
-> hasAutoScrollOnMessageReceived = true if isAtBottom = true in `_onScrollThrottled`.
(keeping in mind to adapt to be "at the end" instead of "at bottom".